delfin/drivers/ibm/storwize_svc/alert_handler.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+from oslo_log import log
+from delfin.common import constants
+from delfin import exception
+from delfin.drivers.ibm.storwize_svc import consts
+
+LOG = log.getLogger(__name__)
+
+
+class Alert_Handler(object):
+    default_me_category = 'storage-subsystem'
+
+    OID_ERR_ID = '1.3.6.1.4.1.2.6.190.4.3'
+    OID_ERR_CODE = '1.3.6.1.4.1.2.6.190.4.4'
+    OID_SEQ_NUMBER = '1.3.6.1.4.1.2.6.190.4.9'
+    OID_LAST_TIME = '1.3.6.1.4.1.2.6.190.4.10'
+    OID_OBJ_TYPE = '1.3.6.1.4.1.2.6.190.4.11'
+    OID_OBJ_ID = '1.3.6.1.4.1.2.6.190.4.12'
+    OID_SYS_NAME = '1.3.6.1.4.1.2.6.190.4.7'
+
+    _mandatory_alert_attributes = (
+        OID_ERR_ID,
+        OID_SEQ_NUMBER,
+        OID_LAST_TIME,
+        OID_OBJ_TYPE,
+        OID_SYS_NAME
+    )
+
+    TIME_PATTERN = '%a %b %d %H:%M:%S %Y'
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def handle_split(split_str, split_char, arr_number):
+        split_value = ''
+        if split_str is not None:
+            tmp_value = split_str.split(split_char, 1)
+            if arr_number == 1 and len(tmp_value) > 1:
+                split_value = tmp_value[arr_number].strip()
+            elif arr_number == 0:
+                split_value = tmp_value[arr_number].strip()
+        return split_value
+
+    @staticmethod
+    def parse_alert(context, alert):
+        for attr in Alert_Handler._mandatory_alert_attributes:
+            if not alert.get(attr):
+                msg = "Mandatory information %s missing in alert message." \
+                      % attr
+                raise exception.InvalidInput(msg)
+
+        try:
+            alert_model = dict()
+            alert_name = Alert_Handler.handle_split(alert.get(
+                Alert_Handler.OID_ERR_ID), ':', 1)
+            error_info = Alert_Handler.handle_split(alert.get(
+                Alert_Handler.OID_ERR_ID), ':', 0)
+            alert_id = Alert_Handler.handle_split(error_info, '=', 1)
+            severity = consts.EVENT_ID_INFO.\
```

Comment:
Severity can be got from snmpTrapOID field which contains any of the following values - tsveETrap, tsveWTrap,  tsveITrap, so we can avoid static mapping list

Reply:
did you check it. it seems there is no severity from trap

Reply:
in a realy trap message,there is no Severity  information,so that i make a static mapping list

Reply:
Yes, snmpTrapOID contains severity field in real trap. Like below
![image](https://user-images.githubusercontent.com/30930862/99631028-e6a91600-2a60-11eb-9785-287f549a934d.png)

Reply:
fixed